dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/ValidationUtils.java

```diff
@@ -177,6 +179,36 @@ public static boolean urlIsValid( String url )
         return new UrlValidator().isValid( url );
     }
 
+    /**
+     * Validates whether a UUID is valid.
+     *
+     * @param uuid the UUID as string.
+     * @return true if the UUID is valid, false otherwise.
+     */
+    public static boolean uuidIsValid( String uuid )
+    {
+        try
+        {
+            UUID.fromString( uuid );
+            return true;
+        }
+        catch ( IllegalArgumentException ex )
+        {
+            return false;
+        }
+    }
+
+    /**
+     * Validates whether a username is valid.
+     *
+     * @param username the username.
+     * @return true if the username is valid, false otherwise.
+     */
+    public static boolean usernameIsValid( String username )
+    {
+        return username != null && username.length() <= UserCredentials.USERNAME_MAX_LENGTH;
```

Comment:
The username string would go directly into the query `from UserCredentials uc where uc.username = :username`. Have we already had sql escape somewhere in the api ? 
e.g. what if username string =  (select username from users order by lastlogin desc limit 1)

Reply:
Good point. We retrieve user credentials in `HibernateUserCredentialsStore` using a Hibernate `Query` with named parameters, where SQL is automatically sanitized.